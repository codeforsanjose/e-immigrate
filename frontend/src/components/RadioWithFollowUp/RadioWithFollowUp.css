.RadioGroup .Radio {
    margin: 0.5em;
    font-size: 1rem;
    display: grid;
    grid-template-columns: min-content auto;
    gap: 0.5rem;
}

.RadioGroup .RadioInput {
    display: flex;
}

.RadioGroup .RadioInput input {
    opacity: 0;
    width: 0;
    height: 0;
}
.RadioGroup .RadioInput input:focus + .RadioControl {
    box-shadow: 0 0px 6px var(--color-overcastBlue);
}

.RadioGroup .RadioInput .RadioButton:checked + .RadioControl {
    background: radial-gradient(
        var(--color-overcastBlue) 50%,
        rgba(255, 0, 0, 0) 51%
    );
}

.RadioGroup .RadioControl {
    display: block;
    width: 1rem;
    height: 1rem;
    border-radius: 50%;
    border: 1px solid var(--color-foggyBlue);
    transform: translateY(-0.05em);
}

.RadioGroup .RadioInput .RadioButton:checked + .RadioControl::before {
    transform: scale(1);
}

.RadioGroup .RadioLabel {
    line-height: 1;
}

.RadioButton[data-touched='true']:invalid:first-of-type ~ .RequiredError {
    display: inline;
    margin-left: 8px;
}

.RequiredError {
    display: none;
    color: var(--color-secondary);
}
