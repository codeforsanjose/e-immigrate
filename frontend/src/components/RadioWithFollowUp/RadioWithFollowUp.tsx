import React from 'react';
import './RadioWithFollowUp.css';
import { ReactSetter } from '../../types/common';
import { CommonComponentAttributes } from '../../types/CommonComponentAttributes';

type RadioWithFollowUpProps = {
    attributes: CommonComponentAttributes & {
        answers?: Array<string>;
        values: Array<string>;
        setShowFollowUp: ReactSetter<Record<string, boolean>>;
    };
};
export function RadioWithFollowUp(props: RadioWithFollowUpProps) {
    const {
        attributes: {
            q,
            answers,
            values,
            bindField,
            setShowFollowUp,
            collectAnswer,
            content,
        },
    } = props;
    return (
        <div className="RadioGroup">
            {answers?.map((option, idx) => (
                <div key={`${q.slug}-${option}`}>
                    <label className="Radio">
                        <span className="RadioInput">
                            <input
                                type="radio"
                                id={`${q.slug}-${option}`}
                                name={q.slug}
                                required={q.required}
                                value={values[idx]}
                                onChange={(e) => {
                                    collectAnswer(q.slug, e.target.value);
                                    values[idx] === 'Yes'
                                        ? setShowFollowUp((prev) => ({
                                            ...prev,
                                            [q.slug]: true,
                                        }))
                                        : setShowFollowUp((prev) => ({
                                            ...prev,
                                            [q.slug]: false,
                                        }));
                                }}
                                className="RadioButton"
                                {...bindField(q.slug)}
                            />
                            <span className="RadioControl"></span>
                        </span>
                        <span className="RadioLabel">{option}</span>
                    </label>
                    <span className="RequiredError">
                            *{content.errorMessage}
                    </span>
                </div>
            ))}
        </div>
    );
};
