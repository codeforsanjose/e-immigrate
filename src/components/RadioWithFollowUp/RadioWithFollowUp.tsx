import React from 'react';
import './RadioWithFollowUp.css';
import { QData, BindFieldFunction, CollectAnswerFunction, ReactSetter } from '../../types/common';

type RadioWithFollowUpProps = {
    attributes: {
        q: QData;
        answers?: Array<string>;
        values: Array<string>;
        bindField: BindFieldFunction;
        collectAnswer: CollectAnswerFunction;
        setShowFollowUp: ReactSetter<Record<string, boolean>>;
        content: {
            errorMessage: string;
        };
    };
};
export function RadioWithFollowUp(props: RadioWithFollowUpProps) {
    const {
        attributes: {
            q,
            answers,
            values,
            bindField,
            setShowFollowUp,
            collectAnswer,
            content,
        },
    } = props;
    return (
        <div className="RadioGroup">
            {answers?.map((option, idx) => (
                <div key={`${q.slug}-${option}`}>
                    <label className="Radio">
                        <span className="RadioInput">
                            <input
                                type="radio"
                                id={`${q.slug}-${option}`}
                                name={q.slug}
                                required={q.required}
                                value={values[idx]}
                                onChange={(e) => {
                                    collectAnswer(q.slug, e.target.value);
                                    values[idx] === 'Yes'
                                        ? setShowFollowUp((prev) => ({
                                            ...prev,
                                            [q.slug]: true,
                                        }))
                                        : setShowFollowUp((prev) => ({
                                            ...prev,
                                            [q.slug]: false,
                                        }));
                                }}
                                className="RadioButton"
                                {...bindField(q.slug)}
                            />
                            <span className="RadioControl"></span>
                        </span>
                        <span className="RadioLabel">{option}</span>
                    </label>
                    <span className="RequiredError">
                            *{content.errorMessage}
                    </span>
                </div>
            ))}
        </div>
    );
};
