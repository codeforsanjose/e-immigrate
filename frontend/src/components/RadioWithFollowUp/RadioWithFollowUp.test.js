import React from 'react';
import RadioWithFollowUp from './RadioWithFollowUp';

import renderer from 'react-test-renderer';

it('RadioWithFollowUp renders correctly', () => {
    const tree = renderer
        .create(
            <RadioWithFollowUp
                q={{}}
                answers={[]}
                values={[]}
                bindField={() => {}}
                showFollowUp={{}}
                setShowFollowUp={{}}
                collectAnswer={{}}
                content={{}}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
