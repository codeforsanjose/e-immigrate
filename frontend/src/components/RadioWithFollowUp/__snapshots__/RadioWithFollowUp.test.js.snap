// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioWithFollowUp renders correctly 1`] = `
<div
  className="RadioGroup"
/>
`;
